samples/nrf_rpc/entropy_nrf53/cpunet/src/entropy_ser.c

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+#include <errno.h>
+#include <init.h>
+#include <drivers/entropy.h>
+
+#include <tinycbor/cbor.h>
+#include <nrf_rpc_cbor.h>
+
+#include <device.h>
+
+#include "../../common_ids.h"
+
+
+#define CBOR_BUF_SIZE 16
+
+enum call_type {
+	CALL_TYPE_STANDARD,
+	CALL_TYPE_CBK,
+	CALL_TYPE_ASYNC,
+};
+
+NRF_RPC_GROUP_DEFINE(entropy_group, "nrf_sample_entropy", NULL, NULL, NULL);
+
+static struct device *entropy;
+
+static void rsp_error_code_send(int err_code)
+{
+	struct nrf_rpc_cbor_ctx ctx;
+
+	NRF_RPC_CBOR_ALLOC(ctx, CBOR_BUF_SIZE);
+
+	cbor_encode_int(&ctx.encoder, err_code);
+
+	nrf_rpc_cbor_rsp_no_err(&ctx);
+}
+
+static void entropy_init_handler(CborValue *packet, void *handler_data)
+{
+	nrf_rpc_cbor_decoding_done(packet);
+
+	entropy = device_get_binding(DT_CHOSEN_ZEPHYR_ENTROPY_LABEL);
+	if (!entropy) {
+		rsp_error_code_send(-NRF_EINVAL);
+		return;
+	}
+
+	rsp_error_code_send(0);
+}
+
+
+NRF_RPC_CBOR_CMD_DECODER(entropy_group, entropy_init, RPC_COMMAND_ENTROPY_INIT,
+			 entropy_init_handler, NULL);
+
+
+static void entropy_get_rsp(int err_code, const uint8_t *data, size_t length)
+{
+	struct nrf_rpc_cbor_ctx ctx;
+
+	NRF_RPC_CBOR_ALLOC(ctx, CBOR_BUF_SIZE + length);
+
+	cbor_encode_int(&ctx.encoder, err_code);
+	cbor_encode_byte_string(&ctx.encoder, data, length);
+
+	nrf_rpc_cbor_rsp_no_err(&ctx);
+}
+
+static void rsp_empty_handler(CborValue *value, void *handler_data)
+{
+}
+
+static void entropy_get_result(int err_code, const uint8_t *data, size_t length,
+			       enum call_type type)
+{
+	struct nrf_rpc_cbor_ctx ctx;
+
+	NRF_RPC_CBOR_ALLOC(ctx, CBOR_BUF_SIZE + length);
+
+	cbor_encode_int(&ctx.encoder, err_code);
+	cbor_encode_byte_string(&ctx.encoder, data, length);
+
+	if (type == CALL_TYPE_ASYNC) {
+		nrf_rpc_cbor_evt_no_err(&entropy_group,
+					RPC_EVENT_ENTROPY_GET_ASYNC_RESULT,
+					&ctx);
+	} else {
+		nrf_rpc_cbor_cmd_no_err(&entropy_group,
+					RPC_COMMAND_ENTROPY_GET_CBK_RESULT,
+					&ctx, rsp_empty_handler, NULL);
+	}
+}
+
+static void entropy_get_handler(CborValue *packet, void *handler_data)
+{
+	CborError cbor_err;
+	int err;
+	int length;
+	uint8_t *buf;
+	enum call_type type = (enum call_type)handler_data;
+
+	cbor_err = cbor_value_get_int(packet, &length);
+
+	nrf_rpc_cbor_decoding_done(packet);
+
+	if (cbor_err != CborNoError || length < 0 || length > 0xFFFF) {
+		err = -NRF_EBADMSG;
+		goto error_exit;
+	}
+
+	buf = k_malloc(length);
```

Comment:
Is this really necessary? We have been avoiding dynamic allocation everywhere we can, and in general Zephyr doesn't use it unless strictly required.

Reply:
It is not strictly required. I will change it.